prov/shm/src/smr_ep.c

```diff
@@ -1070,6 +1088,74 @@ static struct fi_ops_tagged smr_tagged_ops = {
 	.injectdata = fi_no_tagged_injectdata,
 };
 
+#define smr_goto_fast(rma_count) (rma_count == 1)
```

Comment:
I'd kill this define and open code the check directly, especially since it's a simple comparison to 1.

Reply:
The reason I separated this was because we discussed the possibility of expanding the fast path to other factors (not just the rma count but the fast keys for example). Do you want to put this part off until later and just keep the rma_count==1 check? If so, I'll throw this define out and put it straight in the code and adapt it once we have a better idea of what performance looks like and when to chose fast vs. generic.

Reply:
Yes - I'd go ahead of embed this until there's a reason for expanding it